libs/optimizer/src/lib/optimizer.ts

```diff
@@ -0,0 +1,348 @@
+  if (request.track.points.length == 0) {
+    console.warn('Empty track received in optimization request. Returns a 0 score');
+    return ZERO_SCORE;
+  }
+  const originalTrack = request.track;
+  const solverTrack = buildValidTrackForSolver(originalTrack);
+  const flight = toIgcFile(solverTrack);
+  const solverScoringRules = toScoringRules(scoringRules);
+  const options = toOptions(request.options);
+  const solutionIterator = solver(flight, solverScoringRules || {}, options);
+  while (true) {
+    const solution = solutionIterator.next();
+    if (solution.done) {
+      console.debug('solution', JSON.stringify(solution.value, undefined, 2));
+      return toResult(solution.value, originalTrack, solverTrack);
+    }
+    yield toResult(solution.value, originalTrack, solverTrack);
+  }
+}
+
+/**
+ * the solver requires at least 5 points, so if there is not enough points,
+ * we create points between existing ones
+ */
+function buildValidTrackForSolver(track: ScoringTrack) {
+  if (track.points.length >= MIN_POINTS) {
+    return track;
+  }
+  console.debug(`not enough points (${track.points.length}) in track. Interpolate intermediate points`);
+  // avoid potential future bugs with a defensive copy to avoid unwanted mutations on inputs.
+  let newTrack: ScoringTrack = deepCopy(track);
+  while (newTrack.points.length < MIN_POINTS) {
+    const segments: ScoringTrack[] = [];
+    for (let i = 1; i < newTrack.points.length; i++) {
+      // split each segment of the track into two segments
+      segments.push(
+        createSegments(
+          newTrack.points[i - 1],
+          newTrack.points[i],
+          track.startTimeSec,
+          NUM_SEGMENTS_BETWEEN_POINTS,
+          DISTRIBUTION_FACTOR_FOR_ADDED_POINTS,
+        ),
+      );
+    }
+    newTrack = concatTracks(...segments);
+  }
+  console.debug(`new track has ${newTrack.points.length} points`);
+  return newTrack;
+}
+
+
+/**
+ * build a fake igc file from a track, so that the solver can use it
+ * @param track the source track
+ */
+function toIgcFile(track: ScoringTrack): IGCFile {
+  const fixes = track.points.map((point): BRecord => {
+    const timeMilliseconds = point.timeSec * 1000;
+    return {
+      timestamp: timeMilliseconds,
+      time: new Date(timeMilliseconds).toISOString(),
+      latitude: point.lat,
+      longitude: point.lon,
+      valid: true,
+      pressureAltitude: null,
+      gpsAltitude: point.alt,
+      extensions: {},
+      fixAccuracy: null,
+      enl: null,
+    };
+  });
+  // we ignore some properties of the igc-file, as they are not required for the computation
+  // @ts-ignore
+  return {
+    date: new Date(track.startTimeSec * 1000).toISOString(),
+    fixes: fixes,
+  };
+}
+
+function toScoringRules(league: ScoringRules) {
+    return scoringRules.get(league);
+}
+
+type SolverOptions = { maxloop?: number; maxcycle?: number };
+
+function toOptions(options?: OptimizationOptions): SolverOptions {
```

Comment:
toSolverOptions? 
To be consistent with the type

Reply:
Ok. Changed